public/javascripts/signup/signup.js

```diff
@@ -0,0 +1,78 @@
+(function(){
+	//initialize the angular app and inject dependencies.
+	angular.module("olinbaja.signup", ['ngRoute'])
+    .config(function($routeProvider, $locationProvider) {
+      $routeProvider
+        .when('/signup', {
+          templateUrl: './pages/signup.html',
+          controller: 'SignupController',
+          controllerAs: 'vm'
+        })
+
+      $locationProvider.html5Mode({
+        enabled: true,
+        requireBase: false
+      });
+    })
+    .controller('SignupController', SignupController);
+
+  function SignupController($http, $window) {
+    var vm = this;
+
+    $http.get('/auth/isAuthenticated', {}).then(
+      function success(response) {
+        if (response.status == 200) {
+          $window.location = "/";
+        }
+      }, function error(response) {
+      }
+    );
+
+    vm.submit = function(user) {
+      if (!user.email || user.email === '')
```

Comment:
It might be easier/ provide a better user experience to do this validation on fields on the form https://docs.angularjs.org/guide/forms